src/init/starship.bash

```diff
@@ -32,7 +32,11 @@ starship_preexec() {
 # Will be run before the prompt is drawn
 starship_precmd() {
     # Save the status, because commands in this pipeline will change $?
-    STARSHIP_CMD_STATUS=$? STARSHIP_PIPE_STATUS=(${PIPESTATUS[@]})
+    if [[ ${BLE_ATTACHED-} ]]; then
+        STARSHIP_CMD_STATUS=$? STARSHIP_PIPE_STATUS=(${BLE_PIPESTATUS[@]})
```

Comment:
@akinomyoga is this approach fine?

Reply:
It doesn't work because running `[[ ... ]]` washes out both `$?` and `"${PIPESTATUS[@]}"` You need to store those values in the very first command of the precmd function (as in the current master) and then run `STARSHIP_PIPE_STATUS=("${BLE_PIPESTATUS[@]}")` in the case of ble.sh.